include/mbgl/storage/default_file_source.hpp

```diff
@@ -2,9 +2,135 @@
 #define MBGL_STORAGE_DEFAULT_FILE_SOURCE
 
 #include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <vector>
 
 namespace mbgl {
 
+/*
+ * An offline region defined by a style URL, geographic bounding box, zoom range, and
+ * device pixel ratio.
+ */
+class OfflineTilePyramidRegionDefinition {
+public:
+    OfflineTilePyramidRegionDefinition(const std::string&, const LatLngBounds&, double, double, float);
+
+    std::string styleURL;
```

Comment:
An offline region must store _all_ resources necessary for that region. Those resources are necessarily tied to a style: two different styles may use different sources, sprites, and glyphs. Therefore the style is part of the region definition.

Resources that are common between two regions will be deduplicated -- downloaded and stored only once. If you want multiple offline styles, create a region for each one. The result is that union of all required resources will be available offline.